### PR TITLE
fix(Transaction): Force Transaction page to re-render when hash changes

### PR DESCRIPTION
**Pull request by ckniffen**
This also switches the Transaction component to hooks.

## High Level Overview of Change

When the hash changed on the transaction page by either updating the url manually or searching for a transaction using the search box the page would act like it loading it new data but would keep the old transactions information.

By leveraging useEffect the new hash will be used to fetch data.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [ ] Updated files to TypeScript

### CODE REVIEW

File: src/containers/Transactions/index.js [hunk 1]
```diff
@@ -33,12 +34,15 @@ ERROR_MESSAGES.default = {
...
 const getErrorMessage = error => ERROR_MESSAGES[error] || ERROR_MESSAGES.default;
 
-class Transaction extends Component {
-  componentDidMount() {
-    const { t, actions, match, data } = this.props;
+const Transaction = props => {
+  const { identifier = '', tab = 'simple' } = useParams();
+  const match = useRouteMatch();
+  const { t } = useTranslation();
+  const { actions, data, loading } = props;
```
Comment (ledhed2222) [inline src/containers/Transactions/index.js:41]: extreme nit - usually props destructuring happens as the first line in the component. just a pattern. does not block this pr

**Comment by ledhed2222** [comment]
very cool to fix this bug while moving our style forward and while removing lines of code!

**Review by ledhed2222 — APPROVED** [review]

File: src/containers/Transactions/index.js [hunk 2]
```diff
@@ -50,10 +54,9 @@ class Transaction extends Component {
     if (identifier && identifier !== hash) {
       actions.loadTransaction(identifier, rippledSocket);
     }
-  }
+  }, [identifier]);
```
Comment (JST5000) [inline src/containers/Transactions/index.js:57]: Do we also need to include `data` in this dependency array?

Comment (ckniffen) [reply src/containers/Transactions/index.js:57]: No because this is what triggers the data request.  If we specified data it might create a loop.

**Review by JST5000 — APPROVED** [review]
LGTM

**Pull request merged**